pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -965,6 +965,35 @@ func (cm *containerManagerImpl) GetAllocatableMemory() []*podresourcesapi.Contai
 	return containerMemoryFromBlock(cm.memoryManager.GetAllocatableMemory())
 }
 
+func (cm *containerManagerImpl) GetDynamicResources(pod *v1.Pod, container *v1.Container) []*podresourcesapi.DynamicResource {
+	if !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.DynamicResourceAllocation) {
+		return []*podresourcesapi.DynamicResource{}
+	}
+	if !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.KubeletPodResourcesDynamicResources) {
+		return []*podresourcesapi.DynamicResource{}
+	}
+
+	var podDynamicResources []*podresourcesapi.DynamicResource
+	var cdiDevices []*podresourcesapi.CDIDevice
+	podClaimInfoStateList, err := cm.draManager.GetPodClaimInfoState(pod, container)
+	if err != nil {
+		klog.ErrorS(err, "Unable to get pod claim info")
+		return []*podresourcesapi.DynamicResource{}
+	}
+	for _, podClaimInfoState := range podClaimInfoStateList {
+		for _, cdiDevice := range podClaimInfoState.CdiDevices {
+			cdiDevices = append(cdiDevices, &podresourcesapi.CDIDevice{Name: cdiDevice})
```

Comment:
what if we have more than a claim? it seems to me `cdiDevices` is never reset to empty, and it probably should. Could you please check?

Reply:
I think you are right. I will check and update

Reply:
This is going to change with this:
https://github.com/kubernetes/kubernetes/pull/116513

Maybe we should merge that quickly so that this doesn't need to change again.

Reply:
thanks both. This part LGTM now from podresources API perspective.